chapters/solutions.Rmd

```diff
@@ -1582,7 +1654,53 @@ most items would be checked off of the list.
 
 ### Exercise \@ref(packaging-ex-data) {-}
 
-FIXME
+The central idea is that a Python file can find out what directory it was read from,
+so it can look for other files in that directory.
+This means that we don't have assume the package has been installed in a particular place
+or that some environment variable will tell us where to look for our data.
+To test this idea out:
+
+1.  Create a directory called `data-package`.
+
+1.  Put a copy of `collated.csv` in that directory.
+
+1.  Add a file called `collated.py` to that directory.
+
+    ```python
+    '''Example data package.'''
+
+    import os
+    import csv
+
+    def getData():
+        '''Load 'collated.csv' from this file's directory.'''
+        filename = os.path.join(os.path.dirname(__file__), 'collated.csv')
+        with open(filename, 'r') as reader:
+            return [row for row in csv.reader(reader)]
+    ```
+
+1.  Create another file called `test.py`:
+
+    ```python
+    import collated
+
+    data = collated.getData()
+    for row in data:
+        print(row)
+    ```
+
+1.  Run the test program.
```

Comment:
In the solution we probably need to describe the sequence of commands required to create a test environment and then pip install `data-package` into that environment:
```
$ conda create -n data python
$ conda activate data
(data)$ cd data-package
(data)$ pip install -e .
```